### PR TITLE
Stop filling Grading.info consumer_key

### PR DESCRIPTION
**Pull request by marcospri**
For https://github.com/hypothesis/lms/issues/3719

Stop filling the old column now that it can hold nulls.

#### GradingInfo
- `tox -qe dockercompose -- exec postgres psql -U postgres -c "truncate lis_result_sourcedid;"`

- Launch the blackboard assignment `localhost (make devdata) HTML Assignment` as the student user `blackboardstudent2`
https://aunltd-test.blackboard.com/webapps/blackboard/execute/blti/launchPlacement?blti_placement_id=_20_1&content_id=_36_1&course_id=_19_1

- Check the new rows:

```
tox -qe dockercompose -- exec postgres psql -U postgres -c "select oauth_consumer_key, application_instance_id from lis_result_sourcedid;"

 oauth_consumer_key | application_instance_id 
--------------------+-------------------------
                    |                      16
(1 row)
```

- Login now as `blackboardteacher` and launch the assignment again https://aunltd-test.blackboard.com/webapps/blackboard/execute/blti/launchPlacement?blti_placement_id=_20_1&content_id=_36_1&course_id=_19_1

`Blackboard Student 2` should show up in the grading bar student selector.

### CODE REVIEW

File: lms/resources/_js_config/__init__.py [hunk 1]
```diff
@@ -394,7 +394,7 @@ def _get_students(self):
         and had grading info recorded for them.
         """
         grading_infos = self._grading_info_service.get_by_assignment(
-            oauth_consumer_key=self._application_instance.consumer_key,
+            application_instance=self._application_instance,
```
Comment (jon-betts) [inline lms/resources/_js_config/__init__.py:397]: It kind of feels like this is how it maybe should always have worked?

File: lms/services/grading_info.py [hunk 2]
```diff
@@ -23,27 +23,22 @@ def __init__(self, _context, request):
...
         Return all GradingInfo's for a given assignment.
 
         The returned list will contain one GradingInfo for each student who has
         launched this assignment (and had GradingInfo data persisted for them).
 
-        :param oauth_consumer_key: the assignment's oauth_consumer_key
+        :param application_instance: the assignment's application_instance
             (identifies a deployment of our app in an LMS)
-        :type oauth_consumer_key: str
-
```
Comment (jon-betts) [inline lms/services/grading_info.py:36]: :+1: These are just clutter. We don't need so much documentation you can't see the code.

File: tests/unit/lms/services/grading_info_test.py [hunk 3]
```diff
@@ -24,32 +24,9 @@ def test_it(self, svc, matching_grading_infos, application_instance):
...
-                "matching_oauth_consumer_key",
-                "matching_context_id",
-                "other_resource_link_id",
-            ),
-            ("other_oauth_consumer_key", "other_context_id", "other_resource_link_id"),
-        ],
-    )
-    def test_it_returns_an_empty_list_if_there_are_no_matching_GradingInfos(
-        self, svc, oauth_consumer_key, context_id, resource_link_id
-    ):
```
Comment (jon-betts) [inline tests/unit/lms/services/grading_info_test.py:50]: Is something lost here? Previously this was checking each of the items in turn. I'd be interested in your reasoning for removing this, as it does seem like a lot for a little amount of code.

**Review by jon-betts — APPROVED** [review]
Yep this seems pretty simple.

I'm interested to understand the reason the tests were simplified. It seems like we don't have as good coverage as before. I suspect I'm missing something though.

File: tests/unit/lms/services/grading_info_test.py (hunk 3 above)
Comment (marcospri) [reply tests/unit/lms/services/grading_info_test.py:50]: Yeah, can't remember what my reasoning was then (if any). 

Kept the same test using application_instance.

**Pull request merged**